src/constants.js

```diff
@@ -0,0 +1,52 @@
+const charValue = {
+  A:1, B:3, C:3, D:2, E:1, F:4, G:2, H:4, I:1, J:8, K:5, L:1, M:3, N:1, O:1, P:3, Q:10, R:1, S:1, T:1, U:1, V:4, W:4, X:8, Y:4, Z:10
+}
+
+const charFactor = {
+  current: 1,
+  valid: true,
+}
+
+const resetCharFactor = () => {
+  charFactor.current = 1
+  charFactor.valid = true
+  delete charFactor.previous
+}
+
+const getValue = (char) => (!!char.match(/[^A-Za-z]/) ? 0 : charValue[char.toUpperCase()])
```

Comment:
I like this if statement.
Why does it have a double exclamation mark?
What this .match() method does?

Reply:
- The double exlamation mark is to ensure that it gives back either a true or false, even when the result of char.match(/[^A-Za-z]/) is null / undefined, an array or whatever
- the .match() method checks whether something matches a pattern – in this case "does not contain the characters A-Za-z". It's a so called regular expression.